lib/src/main/java/io/cloudquery/helper/ArrowHelper.java

```diff
@@ -71,22 +189,37 @@ public static Schema toArrowSchema(Table table) {
     if (table.getParent() != null) {
       metadata.put(CQ_TABLE_DEPENDS_ON, table.getParent().getName());
     }
+    metadata.put(CQ_EXTENSION_CONSTRAINT_NAME, table.getConstraintName());
     return new Schema(asList(fields), metadata);
   }
 
   public static Table fromArrowSchema(Schema schema) {
     List<Column> columns = new ArrayList<>();
     for (Field field : schema.getFields()) {
-      columns.add(Column.builder().name(field.getName()).type(field.getType()).build());
+      boolean isUnique = field.getMetadata().get(CQ_EXTENSION_UNIQUE) == "true";
```

Comment:
```suggestion
      boolean isUnique = field.getMetadata().get(CQ_EXTENSION_UNIQUE).equals("true");
```

Or use `Objects.equals(field.getMetadata().get(CQ_EXTENSION_UNIQUE), "true")` if it can be `null`.